build.psm1

```diff
@@ -446,6 +446,32 @@ Fix steps:
         Pop-Location
     }
 
+    # publish powershell.config.json
+    # only Windows currently has a default config file that needs to be merged if Preview
+    $version = git --git-dir="$PSSCriptRoot/.git" describe
+    $previewConfigPath = Join-Path -Path $PSScriptRoot -ChildPath "src" -AdditionalChildPath "powershell/powershell-preview.config.json"
```

Comment:
Hmm, I would like to avoid this file if possible. How about getting all experimental features through `Get-ExperimentalFeature`?
After the build finishes, we just run `<publish-folder>\pwsh -noprofile -c Get-ExperimentalFeature | ForEach-Object Name` to get all experimental feature names, and then construct the 'config.json' file.

Maybe remove `src\powershell\powershell.config.json` file too, and dynamically generate `"Microsoft.PowerShell:ExecutionPolicy":"RemoteSigned"` entry for windows as part of the above process.

Reply:
My concern is if there is a case in the future where there are conflicting experimental features the we'd have to be selective on which one is enabled by default

Reply:
Conflicting experimental features probably need to be better handled by the experimental feature infrastructure, as it would be impossible for a user to know as of today if any two experimental features are conflicting. Maybe we should support a way to specify the conflict when registering an experimental feature, so `Get-ExperimentalFeature` can present this information.

Reply:
Thinking about it further, perhaps we can defer that to when such a situation happens.

Reply:
Agreed.

Reply:
There should not be conflicting opportunities - otherwise how would we transfer them to production? _Alternative_ features are more likely, but then they should be like one experimental feature with a switch.

Reply:
It could happen when there are different design proposals for the same feature, and both are implemented as experimental features to find out which one is better.
But that should be very rare in my opinion.